tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -102,11 +108,33 @@ public bool AddRange(ArraySegment<Span> spans)
             {
                 for (int i = 0; i < spans.Count; i++)
                 {
-                    forceKeep |= AddToBuffer(spans.Array[i + spans.Offset]);
+                    AddToBuffer(spans.Array[i + spans.Offset]);
                 }
             }
+        }
+
+        public bool RunSamplers(ArraySegment<Span> trace)
+        {
+            // TODO: Add ability to disable the RareSampler, which the trace agent has
+            // Run the RareSampler early to make sure the signature gets counted
+            var rare = _rareSampler.Sample(trace);
```

Comment:
Given the rare sampler is the only one with state, had we may as well just return immediately if this is true? If so, then FWIW, you could short-circuit here if preferable for readability: 

```csharp
    return _rareSampler.Sample(trace)
        || _prioritySampler.Sample(trace)
        || _errorSampler.Sample(trace)
        || _analyticsEventSampler.Sample(trace);
```

Reply:
What you propose has really great readability. I'll try to do something similar in my coming changes, but note that the RareSampler is the _last_ one to run. It was getting invoked first to mark unique points but its value was only used as the very last fallback

Reply:
OK I made a refactor similar to what you suggest in 69b31153d4033a5c64cf73504aa3dff90bf6a95d . The difference is that the last OR operand is the result of the `_rareSampler.Sample(trace)`, which was run earlier in the method. The reason for this is documented in the code, so hopefully that's clear

Reply:
> OK I made a refactor similar to what you suggest in https://github.com/DataDog/dd-trace-dotnet/commit/69b31153d4033a5c64cf73504aa3dff90bf6a95d
Cool.

> The reason for this is documented in the code, so hopefully that's clear

The reasoning there is clear, thanks. And I understand keeping it like this for consistency with other traceers, but...

> but note that the RareSampler must always run first (unless disabled), but its result is used as the very last fallback

This makes sense to me if the other samplers have side-effects. But as far as I can see, they _don't_. so if the `rareSpanFound` is `true`, the result is always `true`, and running the other samplers is unnecessary?

I'm not passionate about it, especially as if the other samplers _do_ have side effects (or might in the future), then the order is important